### DISCUSSION TITLE
[WASM] Release wasm app crashes and only shows a white page after the splash screen

### DISCUSSION THREAD

**Discussion by tobizk04** [unoplatform/uno#16144] [Q&A] [answered]
Hello everyone.
I have a problem with my wasm uno app in the **release build**.

1. the build is successful
2. start the app without debugger
3. the splash screen is loading and is visible
4. only a white page is visible after loading
5. I think the web assembly crashes

This problem only occurs in the release configuration.
Everything works fine in the debug configuration.

So my questions:
- **Is there any way to detect what is wrong without attach the debugger?**
- I think it is a webassemby problem because the windows app works well also in release build
- Does anyone have any ideas?

Thanks in advance!

**Comment by VRage** [comment]
Did you tried to disable the ILLinker on your Release configuration: `<WasmShellILLinkerEnabled>false</WasmShellILLinkerEnabled>`

**Reply by jeromelaban** [reply]
The linker is a [common cause of such issues](https://platform.uno/docs/articles/features/using-il-linker-webassembly.html#webassembly), you can also enable the logging in release. You can find those settings in the `App.cs` file.

**Reply by tobizk04** [reply]
Thanks for the answer.

- I´ve tried with the ILLinker disabled and then the release is working.
   -> It seems to be an ILLinker problem.


- Then I activated the ILLinker again and I also activated all the logging you mentioned.

-> like this:
```

.UseLogging(configure: (context, logBuilder) =>
{
		// Configure log levels for different categories of logging
		logBuilder
			.SetMinimumLevel(LogLevel.Information)

			// Default filters for core Uno Platform namespaces
			.CoreLogLevel(LogLevel.Warning);

		// Uno Platform namespace filter groups
		// Uncomment individual methods to see more detailed logging
		//// Generic Xaml events
		logBuilder.XamlLogLevel(LogLevel.Debug);
		//// Layout specific messages
		logBuilder.XamlLayoutLogLevel(LogLevel.Debug);
		//// Storage messages
		logBuilder.StorageLogLevel(LogLevel.Debug);
		//// Binding related messages
		logBuilder.XamlBindingLogLevel(LogLevel.Debug);
		//// Binder memory references tracking
		logBuilder.BinderMemoryReferenceLogLevel(LogLevel.Debug);
		//// Debug JS interop
		logBuilder.WebAssemblyLogLevel(LogLevel.Debug);

}, enableUnoLogging: true)


```

But the log messages do not tell me anything about the origin of the error.
I have no idea which lines of code cause the error.

Here are the logs:
[uno_crash_logs.txt](https://github.com/unoplatform/uno/files/14931397/uno_crash_logs.txt)

**Are there any other ways to identify the error?**

**Reply by VRage** [reply, marked as answer]
I think he meant: 
```
        <MonoRuntimeDebuggerEnabled>true</MonoRuntimeDebuggerEnabled>
        <DefineConstants>$(DefineConstants);TRACE;DEBUG</DefineConstants>
        <DebugType>portable</DebugType>
        <DebugSymbols>true</DebugSymbols> 
```
on your Release configuration. If enabled your should see errors or warnings in your VS console

**Reply by jeromelaban** [reply]
@tobizk04 It is likely that some operation in the code you've created is causing an exception that is not reported to the logs. You can add additional logs in your app to determine coarsely where the app is failing.

**Comment by tobizk04** [comment]
I found the issue.
-> it was a ILLinker problem
-> I solved it with additional entries in the LinkerConfig.xml

Thanks! :-)

**Reply by jeromelaban** [reply]
Thanks for the update!
